samza-core/src/main/java/org/apache/samza/operators/StreamGraphImpl.java

```diff
@@ -186,8 +193,26 @@ public ContextManager getContextManager() {
     return this.contextManager;
   }
 
-  /* package private */ int getNextOpId() {
-    return this.opId++;
+  /**
+   * Gets the unique ID for the next operator in the graph. The ID is of the following format:
+   * jobName-jobId-opCode-(opName|nextOpNum);
+   *
+   * @param opCode the {@link OpCode} of the next operator
+   * @param opName the optional user-provided name of the next operator or null
+   * @return the unique ID for the next operator in the graph
+   */
+  /* package private */ String getNextOpId(OpCode opCode, String opName) {
+    String nextOpId = String.format("%s-%s-%s-%s",
+        config.get(JobConfig.JOB_NAME()),
```

Comment:
Shouldn't this be appName? If we create multi-job applications later, this will mandate migration of state/streams.